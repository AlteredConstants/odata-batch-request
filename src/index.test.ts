import { ODataBatch, ODataBatchOperation, Method } from "./index"

test("GET only", () => {
  const batch = new ODataBatch("host/service", [
    new ODataBatchOperation(Method.Get, "Customers('ALFKI')"),
    new ODataBatchOperation(Method.Get, "Products"),
  ])
  expect(batch.value).toMatchSnapshot()
})

test("GET and POST", () => {
  const batch = new ODataBatch("host/service", [
    new ODataBatchOperation(Method.Get, "Customers('ALFKI')"),
    new ODataBatchOperation(Method.Post, "Customers", {
      headers: { "Content-Type": "application/atom+xml;type=entry" },
      body: "<AtomPub representation of a new Customer>",
    }),
  ])
  expect(batch.value).toMatchSnapshot()
})
