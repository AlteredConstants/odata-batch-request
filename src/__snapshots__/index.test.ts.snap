// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET and POST 1`] = `
"POST host/service/$batch HTTP/1.1
Content-Type: multipart/mixed; boundary=batch_36522ad7-fc75-4b56-8c71-56071383e77b

--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

GET Customers('ALFKI') HTTP/1.1


--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

POST Customers HTTP/1.1
Content-Type: application/atom+xml;type=entry

<AtomPub representation of a new Customer>
--batch_36522ad7-fc75-4b56-8c71-56071383e77b--"
`;

exports[`GET only 1`] = `
"POST host/service/$batch HTTP/1.1
Content-Type: multipart/mixed; boundary=batch_36522ad7-fc75-4b56-8c71-56071383e77b

--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

GET Customers('ALFKI') HTTP/1.1


--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

GET Products HTTP/1.1


--batch_36522ad7-fc75-4b56-8c71-56071383e77b--"
`;
